frontend/src/components/pod/List.tsx

```diff
@@ -21,9 +22,15 @@ export function makePodStatusLabel(pod: Pod) {
 
   return <StatusLabel status={status}>{phase}</StatusLabel>;
 }
+interface PodListProps {
+  pods?: Pod[];
+  error?: ApiError | null;
+}
+export default function PodList(props: PodListProps) {
+  const { pods, error } = props;
+  const [apiPods, apiError] = Pod.useList();
```

Comment:
This means we'll always attempt to fetch the pod list, regardless of whether the pods are passed to it.
I think maybe we could just split out the PodList in PodListRenderer (this one accepts pods as props) and PodList (this one uses Pod.useList and passes the result to a PodListRenderer). This way you can call PodListRenderer in the deployment details.